modules/storage.js

```diff
@@ -188,4 +178,54 @@ module.exports = class Storage {
   getAstNodesData(uri) {
     return this.astNodesData.get(uri);
   }
+
+  getRuleTempId(ruleId) {
+    return this.ruleTempId.get(ruleId);
+  }
+
+  setRuleTempId(ruleId, ruleTempId) {
+    this.ruleTempId.set(ruleId, ruleTempId);
+  }
+
+  removeRuleTempId(ruleId) {
+    this.ruleTempId.delete(ruleId);
+  }
+
+  setRuleTempIdToTestCase(testCaseStartedId, ruleTempId) {
+    this.ruleTempIdToTestCaseStartedId.set(testCaseStartedId, ruleTempId);
+  }
+
+  getRuleTempIdToTestCase(testCaseStartedId) {
+    return this.ruleTempIdToTestCaseStartedId.get(testCaseStartedId);
+  }
+
+  removeRuleTempIdToTestCase(testCaseStartedId) {
+    this.ruleTempIdToTestCaseStartedId.delete(testCaseStartedId);
+  }
+
+  getStartedChildren(ruleTempId) {
+    return this.startedChildren.get(ruleTempId) || [];
```

Comment:
Do we hold `startedChildren` only for rules? If yes, please use meaningful naming here to mention it.